verify-cborld-presentation/src/pages/result.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2019 - MATTR Limited
+ * All rights reserved
+ * Confidential and proprietary
+ */
+
+import { Box, Link, StylesProvider } from "@material-ui/core";
+import { encode as base64Encode, decode as base64Decode } from "@stablelib/base64";
+import base32Decode from "base32-decode";
+import Lottie from "lottie-react";
+import { inflate } from "pako";
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+
+import errorAnimation from "../assets/lotti/error.json";
+import loadingSpinnerAnimation from "../assets/lotti/loadingSpinner.json";
+import successAnimation from "../assets/lotti/success.json";
+import { Credential, GenericButton } from "../components";
+import type { Credential as CredentialType } from "../service/types";
+import { isCredential } from "../service/types/validation/credential";
+
+type TrustedIssuer = {
+  did: string;
+  domain: string;
+};
+const trustedIssuers: TrustedIssuer[] = [
+  // example issuer
+  // {
+  //   did: "did:key:z6Mkne6kfQ2bY396dKTuraGLxiDoBrXXjVvH2Zv7nA1MnDoM",
+  //   domain: "localhost:3000",
+  // },
+];
+export interface ResultPageProps {
+  qrCode: string | undefined;
+}
+
+type VerifiationStatus = "checking" | "failed" | "success";
+
+export const ResultPage: React.FC<ResultPageProps> = (props: ResultPageProps) => {
+  const [credential, setCredential] = useState<CredentialType>();
+  const [trustedIssuer, setTrustedIssuer] = useState<string>();
+  const [verificationState, setVerificationState] = useState<VerifiationStatus>("checking");
+
+  useEffect(() => {
+    const qrCode = props.qrCode;
+    if (qrCode) {
+      (async () => {
+        const base64EncodedCborData = base64Encode(inflate(new Uint8Array(base32Decode(qrCode, "RFC4648"))));
```

Comment:
This method can be extracted out?

Reply:
Fair enough, have moved to seperate file in service folder

Reply:
Sorry I mean the whole 
```
// Decode cbor using platform
      const fetchResult = await fetch(`/core/v1/linkeddata/convert`, {
        method: "post",
        headers: {
          "Content-type": "application/json",
        },
        body: JSON.stringify({
          options: {
            inputFormat: "cborld",
            outputFormat: "jsonld",
            outputEncoding: "base64",
          },
          data: base64Encode(decodedQrData),
        }),
      });

      if (!fetchResult.ok) {
        console.log("could not convert cbor data", fetchResult.body);
        return setVerificationState("failed");
      }
      const { data } = await fetchResult.json();
      const decodedJsonld = JSON.parse(Buffer.from(base64Decode(data)).toString());

      // Extract credential
      const credential = decodedJsonld.verifiableCredential[0];
      if (!isCredential(credential)) {
        console.log("credential failed validation", credential);
        return setVerificationState("failed");
      }
      setCredential(credential);

      // use platform to verify the presentation
      const verifyResult = await fetch(`/core/v1/presentations/verify`, {
        method: "post",
        headers: {
          "Content-type": "application/json",
        },
        body: JSON.stringify({
          presentation: decodedJsonld,
        }),
      });

      const { verified } = await verifyResult.json();
      if (!verified) {
        console.log("failed to verify presentation", decodedJsonld);
        return setVerificationState("failed");
      }

      // Check if trusted issuer
      const foundTrustedIssuer = trustedIssuers.find(({ did }) => credential.issuer.id === did);
      if (!foundTrustedIssuer) {
        console.log("credential issuer is not part of the trusted issuer list", credential.issuer.id);
        return setVerificationState("failed");
      }
```